tests/unittests/GraphOptzTest.cpp

```diff
@@ -527,6 +527,46 @@ TEST_F(GraphOptz, sinkTransposeBelowOptimizeBatchNormWithPredicate) {
   EXPECT_EQ(F_->getNodes().size(), 3);
 }
 
+/// Sinks Transpose below Rescale potentially exposing futher optimizations.
+/// For example folding Rescale in to Convolution.
+TEST_F(GraphOptz, sinkTransposeBelowRescale) {
+  // Inputs.
+  const size_t origDims[] = {1, 5, 10, 15};
+  const size_t transposedDims[] = {1, 15, 5, 10};
+  auto *input = mod_.createPlaceholder(ElemKind::Int8QTy, origDims, 0.1, 0,
+                                       "input", false);
+  auto *filter = mod_.createPlaceholder(ElemKind::Int8QTy, {15, 1, 1, 3}, 0.1,
+                                        0, "filter", false);
+  auto *bias =
+      mod_.createPlaceholder(ElemKind::Int32QTy, {15}, 0.01, 0, "bias", false);
+
+  // Graph.
+  ConvolutionNode *conv =
+      F_->addNode(new ConvolutionNode("conv", input->getType(), input, filter,
+                                      bias, {1, 1}, {1, 1}, {0, 0, 0, 0}, 1));
+
+  auto *T = F_->createTranspose("transpose", conv, NHWC2NCHW);
+  auto *RT = mod_.uniqueType(ElemKind::Int8QTy, T->getResult().dims(), 0.2, 0);
+  auto *R = F_->createRescaleQuantized("rescale", T, RT);
+  SaveNode *O = F_->createSave("ret", R);
+
+  EXPECT_EQ(F_->getNodes().size(), 4);
+  EXPECT_EQ(RT->dims(), llvm::makeArrayRef(transposedDims));
+
+  ::glow::optimize(F_, CompilationMode::Infer);
+
+  // Expecting Transpose->Output rather than Rescale->Output.
+  auto *transpose = llvm::dyn_cast<TransposeNode>(O->getInput());
+  ASSERT_NE(transpose, nullptr);
+  ASSERT_TRUE(llvm::isa<ConvolutionNode>(transpose->getInput()));
+  conv = llvm::cast<ConvolutionNode>(transpose->getInput());
```

Comment:
nit: please do something like `auto convNV = transpose->getInput();`, and then below you can do `convNV.dims()` instead of using an explicit output number with `conv->dims(0)`.